### PR TITLE
PDI-2056: Add Defaults to Usage Output

### PR DESCRIPTION
**Pull request by erikostien-pingidentity**
- Update all PFlag structs to remove DefValue and manually add default information to Usage field.
- Also remove config command logic.
- Change --color flag to --no-color and update all logic and testing for the flag.
- Update all boolean flags to be PFlag configured boolean flags.

### CODE REVIEW

**Comment by patrickcping** [comment]
Potential problem:
```
➜  pingcli git:(PDI-2056) go run ./ completion zsh
Failed to validate Ping CLI configuration - Failure
Fatal: failed to validate Ping CLI configuration: invalid configuration key(s) found in profile pingfederate-terraform-uat: color
Must use one of: activeProfile, description, export.format, export.outputDirectory, export.overwrite, export.pingone.environmentID, export.services, noColor, outputFormat, request.accessToken, request.accessTokenExpiry, request.service, service.pingfederate.adminAPIPath, service.pingfederate.authentication.accessTokenAuth.accessToken, service.pingfederate.authentication.basicAuth.password, service.pingfederate.authentication.basicAuth.username, service.pingfederate.authentication.clientCredentialsAuth.clientID, service.pingfederate.authentication.clientCredentialsAuth.clientSecret, service.pingfederate.authentication.clientCredentialsAuth.scopes, service.pingfederate.authentication.clientCredentialsAuth.tokenURL, service.pingfederate.authentication.type, service.pingfederate.caCertificatePemFiles, service.pingfederate.httpsHost, service.pingfederate.insecureTrustAllTLS, service.pingfederate.xBypassExternalValidationHeader, service.pingone.authentication.type, service.pingone.authentication.worker.clientID, service.pingone.authentication.worker.clientSecret, service.pingone.authentication.worker.environmentID, service.pingone.regionCode
exit status 1
```

This probably needs something to remove `color` from all profiles when `nocolor` is set, or, the param switch is `--no-color` but the config file remains `color`

**Comment by patrickcping** [comment]
Is there a possibility of removing the `--profile-name` param on `config set` and `config get` commands, as `--profile` could do the same job?

**Comment by erikostien-pingidentity** [comment]
> Potential problem:
> 
> ```
> ➜  pingcli git:(PDI-2056) go run ./ completion zsh
> Failed to validate Ping CLI configuration - Failure
> Fatal: failed to validate Ping CLI configuration: invalid configuration key(s) found in profile pingfederate-terraform-uat: color
> Must use one of: activeProfile, description, export.format, export.outputDirectory, export.overwrite, export.pingone.environmentID, export.services, noColor, outputFormat, request.accessToken, request.accessTokenExpiry, request.service, service.pingfederate.adminAPIPath, service.pingfederate.authentication.accessTokenAuth.accessToken, service.pingfederate.authentication.basicAuth.password, service.pingfederate.authentication.basicAuth.username, service.pingfederate.authentication.clientCredentialsAuth.clientID, service.pingfederate.authentication.clientCredentialsAuth.clientSecret, service.pingfederate.authentication.clientCredentialsAuth.scopes, service.pingfederate.authentication.clientCredentialsAuth.tokenURL, service.pingfederate.authentication.type, service.pingfederate.caCertificatePemFiles, service.pingfederate.httpsHost, service.pingfederate.insecureTrustAllTLS, service.pingfederate.xBypassExternalValidationHeader, service.pingone.authentication.type, service.pingone.authentication.worker.clientID, service.pingone.authentication.worker.clientSecret, service.pingone.authentication.worker.environmentID, service.pingone.regionCode
> exit status 1
> ```
> 
> This probably needs something to remove `color` from all profiles when `nocolor` is set, or, the param switch is `--no-color` but the config file remains `color`

This is simply a validation step on pingcli startup. It informs the user that their config file is using a config key that is not recognized/does not exist. While this is a breaking change, since this is before beta, I think this is fine to add in without additional migration logic.

**Comment by erikostien-pingidentity** [comment]
> Is there a possibility of removing the `--profile-name` param on `config set` and `config get` commands, as `--profile` could do the same job?

This is possible, but it will remove a subtlety here. For example in the following niche case, if a user specifies '--profile A' to output tool responses as json, but the user wants to modify --profile-name B, this would no longer be possible without additionally using the --output-format flag.

Since this is niche, I don't think there is an issue removing this capability.
@patrickcping Thoughts?

**Comment by patrickcping** [comment]
> > Is there a possibility of removing the `--profile-name` param on `config set` and `config get` commands, as `--profile` could do the same job?
> 
> This is possible, but it will remove a subtlety here. For example in the following niche case, if a user specifies '--profile A' to output tool responses as json, but the user wants to modify --profile-name B, this would no longer be possible without additionally using the --output-format flag.
> 
> Since this is niche, I don't think there is an issue removing this capability. @patrickcping Thoughts?

I didn't consider that case, good spot.  I agree it is niche though, let's flatten to `--profile` and remove that capability, on the basis that if users are setting a profile value on profile "A", they're also invoking the runtime config of that profile at the same time

File: internal/configuration/config/add_profile.go [hunk 1]
```diff
@@ -69,9 +67,10 @@ func initAddProfileSetActiveOption() {
 		Flag: &pflag.Flag{
 			Name:      cobraParamName,
 			Shorthand: "s",
-			Usage:     "Set the new configuration profile as the active profile.",
-			Value:     cobraValue,
-			DefValue:  "false",
+			Usage: "Set the new configuration profile as the active profile. " +
+				"(default false)",
+			Value:       cobraValue,
+			NoOptDefVal: "true", // Make this flag a boolean flag
```
[the remark this answers is not in the document]
Comment (erikostien-pingidentity) [reply internal/configuration/config/add_profile.go:73]: This is the default value when specified like `--set-active`, whereas if it is not specified, the default value is false.

**Pull request merged**